src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Serialization/ObjectWriter.cs

```diff
@@ -129,6 +129,19 @@ public void Dispose()
     public void WriteUInt16(ushort value) => _writer.Write(value);
     public void WriteString(string? value) => WriteStringValue(value);
 
+    public Stream BaseStream => _writer.BaseStream;
+
+    public void Reset(bool includeValidationBytes = true)
+    {
+        _stringReferenceMap.Reset();
+        _writer.BaseStream.Position = includeValidationBytes ? 2 : 0;
```

Comment:
might want to doc this.  perhaps that the bytes are already there, and we want to move after it.

alterantively, hard reset to 0 and write the bytes again.